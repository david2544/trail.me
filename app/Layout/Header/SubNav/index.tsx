import React from 'react';
import { Link } from 'react-router-dom';
import classnames from 'classnames';
import styles from './styles.module.scss';

interface SubNavProps {
  subMenuItem: {
    link: string;
    title?: string;
    titleSub?: string;
    icon?: string;
    description: string;
    isExternal?: boolean;
  };
}

const SubNav: React.FC<SubNavProps> = ({ subMenuItem }) =>
  subMenuItem.isExternal ? (
    <li className={styles.subNavExternal}>
      <div className={styles.subNavItem}>
        <a
          href={subMenuItem.link}
          className={styles.link}
          target="_blank"
          rel="noopener noreferrer"
        >
          <div className={styles.menuTitle}>
            <div className={styles.menuTitleMain}>{subMenuItem.title}</div>
            <div className={styles.menuTitleSub}>{subMenuItem.titleSub}</div>
          </div>
          <div className={styles.menuDescription}>{subMenuItem.description}</div>
        </a>
      </div>
    </li>
  ) : (
    <li className={styles.subNavItem}>
      <Link to={subMenuItem.link} className={styles.link}>
        <div className={styles.menuTitle}>{subMenuItem.title}</div>
        <div
          className={classnames(styles.menuDescription, {
            [styles.menuDescriptionSpacing]: !!subMenuItem.title,
          })}
        >
          {subMenuItem.description}
        </div>
      </Link>
    </li>
  );

export default SubNav;
