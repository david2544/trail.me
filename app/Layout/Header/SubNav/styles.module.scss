@media (max-width: 767px) {
  .subNavItem {
    padding-top: 18px;
    display: block;
    color: black;
    float: none;
  }
}

.link {
  padding: 0;
  display: block;
  color: inherit;
}

@media (min-width: 768px) {
  .menuTitle {
    display: inline-block;
    font-size: 20px;
    position: absolute;
    width: 100px;
    color: black;
  }

  .link {
    color: black;
    border-bottom: 1px solid rgba(0, 0, 0, 0.1);
    padding: 12px 0;
    position: relative;
    text-decoration: none;
  }
}

.menuLogo {
  display: inline-block;
  font-size: 20px;
  position: absolute;
  width: 100px;

  @media (max-width: 767px) {
    display: none;
  }
}

.menuDescription {
  display: inline-block;
  text-decoration: inherit;
}

.menuDescriptionSpacing {
  margin-left: 110px;
  padding-left: 12px;

  @media (max-width: 767px) {
    display: none;
  }
}

.subNavExternal {
  margin-top: 3px;
  font-weight: 300;
  list-style: none;

  @media (min-width: 768px) {
    .menuTitle {
      display: inline-block;
      width: 100px;
      position: absolute;
      top: 5px;
    }
  }

  .menuTitleMain {
    font-size: 14px;
  }

  .menuTitleSub {
    font-size: 10px;
  }

  .menuDescription {
    width: 300px;
    margin-left: 110px;
    padding-left: 12px;

    @media (max-width: 767px) {
      display: none;
    }
  }
}
