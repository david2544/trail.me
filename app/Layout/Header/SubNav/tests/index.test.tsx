import React from 'react';
import { shallow } from 'enzyme';
import SubNav from '../index';

describe('<SubNav />', () => {
  it('should match snapshot', () => {
    const props = {
      subMenuItem: {
        link: 'some string',
        title: 'some string',
        description: 'some string',
        isExternal: false,
      },
    };

    const wrapper = shallow(<SubNav {...props} />);

    expect(wrapper).toMatchSnapshot();
  });

  it('should match snapshot', () => {
    const props = {
      subMenuItem: {
        link: 'some string',
        title: 'some string',
        titleSub: 'some string',
        description: 'some string',
        isExternal: true,
      },
    };

    const wrapper = shallow(<SubNav {...props} />);

    expect(wrapper).toMatchSnapshot();
  });
});
