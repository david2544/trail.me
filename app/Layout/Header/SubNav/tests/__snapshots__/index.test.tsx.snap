// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SubNav /> should match snapshot 1`] = `
<li
  className="subNavItem"
>
  <Link
    className="link"
    to="some string"
  >
    <div
      className="menuTitle"
    >
      some string
    </div>
    <div
      className="menuDescription menuDescriptionSpacing"
    >
      some string
    </div>
  </Link>
</li>
`;

exports[`<SubNav /> should match snapshot 2`] = `
<li
  className="subNavExternal"
>
  <div
    className="subNavItem"
  >
    <a
      className="link"
      href="some string"
      rel="noopener noreferrer"
      target="_blank"
    >
      <div
        className="menuTitle"
      >
        <div
          className="menuTitleMain"
        >
          some string
        </div>
        <div
          className="menuTitleSub"
        >
          some string
        </div>
      </div>
      <div
        className="menuDescription"
      >
        some string
      </div>
    </a>
  </div>
</li>
`;
